File: web/my-financial-hub/src/commom/interfaces/transaction.ts
import { Account } from './account';
import { Category } from './category';

export interface Transaction{
  id?: string,
  description: string,
  amount: number,

  finishDate: string,
  targetDate: string,

  accountId: string,
  account?: Account,
  
  categoryId: string,
  category?: Category,

  isActive: boolean,

  type : TransactionType,
  status : TransactionStatus
}

export enum TransactionType{
  Earn = 1,
  Expense = 2
}

export enum TransactionStatus
{
  NotCommitted = 0,
  Committed = 1
}

export const defaultTransaction: Transaction = {
  description: '',
  amount: 1,

  finishDate: new Date().toISOString().split('T')[0],
  targetDate: new Date().toISOString().split('T')[0],

  accountId: '',
  account: undefined,
  
  categoryId: '',
  category: undefined,

  isActive: true,

  type : 1,
  status : 0 
};

File: src/api/core/FinancialHub.Core.Domain.Balances/Models/Transaction.cs
using FinancialHub.Common.Models;
using FinancialHub.Core.Domain.Balances.Enums;

namespace FinancialHub.Core.Domain.Balances.Models
{
    public class Transaction : BaseModel
    {
        public string Description { get; private set; }
        public decimal Amount { get; private set; }

        public DateTimeOffset TargetDate { get; private set; }
        public DateTimeOffset FinishDate { get; private set; }

        public Category Category { get; private set; }

        public bool IsActive { get; private set; }

        public TransactionStatus Status { get; private set; }
        public TransactionType Type { get ; private set; }

        public bool IsPaid => this.IsActive && this.Status == TransactionStatus.Committed;
    }
}
